x-pack/plugins/maps/public/classes/sources/esql_source/create_source_editor.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { EuiSkeletonText } from '@elastic/eui';
+import { ES_GEO_FIELD_TYPE } from '../../../../common/constants';
+import type { ESQLSourceDescriptor } from '../../../../common/descriptor_types';
+import { getIndexPatternService } from '../../../kibana_services';
+import { ESQLEditor } from './esql_editor';
+import { ESQL_GEO_POINT_TYPE } from './esql_utils';
+
+interface Props {
+  onSourceConfigChange: (sourceConfig: Partial<ESQLSourceDescriptor> | null) => void;
+}
+
+export function CreateSourceEditor(props: Props) {
+  const [isInitialized, setIsInitialized] = useState(false);
+  const [esql, setEsql] = useState('');
+  const [dateField, setDateField] = useState<string | undefined>();
+
+  useEffect(() => {
+    let ignore = false;
+    getIndexPatternService()
+      .getDefaultDataView()
+      .then((defaultDataView) => {
+        if (ignore) {
+          return;
+        }
+
+        if (defaultDataView) {
+          let geoField: string | undefined;
+          const initialDateFields: string[] = [];
+          for (let i = 0; i < defaultDataView.fields.length; i++) {
+            const field = defaultDataView.fields[i];
+            if (!geoField && ES_GEO_FIELD_TYPE.GEO_POINT === field.type) {
+              geoField = field.name;
+            } else if ('date' === field.type) {
+              initialDateFields.push(field.name);
+            }
+          }
+
+          if (geoField) {
+            let initialDateField: string | undefined;
+            if (defaultDataView.timeFieldName) {
+              initialDateField = defaultDataView.timeFieldName;
+            } else if (initialDateFields.length) {
+              initialDateField = initialDateFields[0];
+            }
+            const initialEsql = `from ${defaultDataView.getIndexPattern()} | KEEP ${geoField} | limit 10000`;
```

Comment:
If a map already has other layers, would it be possible to guess which index to use here instead of using a default one?

Reply:
Nice idea for a more relevant ES|QL statement. I have added this with https://github.com/elastic/kibana/pull/173481/commits/9be32b37b4a409c69cde52d8b0744ff86d401a77

Reply:
Awesome! Thanks!